clients/interfaces/builder.go

```diff
@@ -0,0 +1,26 @@
+/*******************************************************************************
+ * Copyright 2020 Dell Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+
+package interfaces
+
+// RestClientBuilder is the interface for types that need to define some way to retrieve state information about services.
+// This information can be anything that must be determined at runtime, whether it is unknown or simply not yet known.
+type RestClientBuilder interface {
```

Comment:
Name is ambiguous to me.  "builder" is a relatively generic term.  Could be mistaken for the GoF Builder pattern (which it most definitely isn't a specific implementation of).

Could this be simply a URL interface?